test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_default_config_test.clj

```diff
@@ -0,0 +1,67 @@
+(ns puppetlabs.trapperkeeper.services.webserver.jetty9-default-config-test
+  "
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; VALIDATION OF DEFAULT JETTY CONFIGURATION VALUES
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+NOTE: IF A TEST IN THIS NAMESPACE FAILS, AND YOU ALTER THE VALUE TO MAKE IT
+PASS, IT IS YOUR RESPONSIBILITY TO DOUBLE-CHECK THE DOCS TO SEE IF THERE
+IS ANYWHERE IN THEM THAT THE NEW VALUE NEEDS TO BE ADDED.
+
+This namespace is a little different than most of our test namespaces.  It's
+not really intended to test any of our own code, it's just here to provide
+us with a warning in the event that Jetty changes any of the default
+configuration values.
+
+In the conversation leading up to https://tickets.puppetlabs.com/browse/TK-168
+we decided that it was generally not a good idea to be hard-coding our own
+default values for the settings that we exposed, and that it would be a better
+idea to allow Jetty to use its implicit default values for any settings that
+are not explicitly set in a TK config file.  Otherwise, we're at risk of
+the Jetty authors coming up with a really compelling reason to change a
+default value between releases, and us not picking up that change.
+
+Therefore, we decided that all the settings we expose should just fall
+through to Jetty's implicit defaults, and that individual TK application
+authors can override any appropriate settings in their packaging if needed.
+
+However, there was some concern that if an upstream Jetty default were to
+change without us knowing about it, it could have other implications for our
+applications that we ought to be aware of.  Therefore, we agreed that it
+would be best if we had some way of making sure we could identify when
+that situation arose.
+
+That is the purpose of this namespace.  It basically provides assertions
+to validate that we know what Jetty's implicit default value is for all of
+the settings we expose.  If we bump to a new version of Jetty in the future
+and any of these implicit defaults have changed, these tests will fail.  If
+that happens, we can attempt to evaluate the impact of the change and
+react accordingly."
+  (:require [clojure.test :refer :all]
+            [schema.test :as schema-test])
+  (:import (org.eclipse.jetty.server HttpConfiguration ServerConnector Server)
+           (org.eclipse.jetty.client HttpClient)))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+(deftest default-request-header-size-test
+  (let [http-config (HttpConfiguration.)]
+    (is (= 8192 (.getRequestHeaderSize http-config)))))
+
+(deftest default-jetty-http-client-settings-test
```

Comment:
Presumably, these tests are here to validate the defaults that the ProxyServlet uses for the HttpClient it creates.  I don't think that will work, though, because the ProxyServlet applies its own defaults over the top of the ones that the HttpClient constructor creates.  See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.9.v20150224/jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java#L268-L271 and https://github.com/eclipse/jetty.project/blob/jetty-9.2.9.v20150224/jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java#L278-L280.  This may work okay for the `requestBufferSize` for now since the ProxyServlet doesn't change the default but it doesn't appear that this will work for `idleTimeout` because the default is changed to 30 seconds.

We may need to have these tests construct a ProxyServlet and read the defaults back from what the `createHttpClient()` method would return - or just read back the _client directly (although that's a private member variable of the AbstractProxyServlet class so we'd probably have to do some ugly reflection to do that).